### PR TITLE
common/Formatter: don't construct string if string_view work

### PR DESCRIPTION
**Pull request by changchengx**
Signed-off-by: Changcheng Liu <changcheng.liu@aliyun.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/Formatter.cc [hunk 1]
```diff
@@ -85,9 +85,7 @@ Formatter *Formatter::create(std::string_view type,
 			     std::string_view default_type,
 			     std::string_view fallback)
 {
-  std::string mytype(type);
-  if (mytype == "")
-    mytype = default_type;
+  std::string_view mytype = type == "" ? default_type : type;
```
Comment (tchaikov) [inline src/common/Formatter.cc:88]:
```suggestion
  std::string_view mytype(type);
  if (mtype.empty()) {
    mytype = default_type;
  }
```

Comment (changchengx) [reply src/common/Formatter.cc]: Done

**Comment by changchengx** [comment]
test this please

**Comment by changchengx** [comment]
jenkins, test this please

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-08-18_13:00:33-rados-wip-kefu-testing-2020-08-18-1658-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-19_06:47:30-rados-wip-kefu-testing-2020-08-19-1141-distro-basic-smithi/

**Pull request merged**